tests/App/Admin/TextEntityAdmin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\FormatterBundle\Tests\App\Admin;
+
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\FormatterBundle\Form\Type\FormatterType;
+use Sonata\FormatterBundle\Form\Type\SimpleFormatterType;
+
+/**
+ * @extends AbstractAdmin<TextEntity>
+ */
+final class TextEntityAdmin extends AbstractAdmin
+{
+    protected function configureFormFields(FormMapper $form): void
+    {
+        $form
+            ->add('simpleText', SimpleFormatterType::class, [
+                'format' => 'rawhtml',
+            ])
+            ->add('text', FormatterType::class, [
+                'listener' => false,
```

Comment:
If I use listener => true then I have to pass a event Dispatcher. The docs says this: https://docs.sonata-project.org/projects/SonataFormatterBundle/en/4.x/reference/formatter_widget/#how-to-use-it

Somehow the `$form` is suposed to have a `getEventDispatcher()`, but I don't see it.

@VincentLanglet do you know how to get an EventDispatcher here? Or do I have to inject one on the construct?

Reply:
It's not `$form->getEventDispatcher()` but `$formBuilder->getEventDispatcher()`.
https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Form/FormConfigBuilder.php#L182

So you can do `$form->getFormBuilder()->getEventDispatcher()`

Reply:
Fixed.